### PR TITLE
Fix floating point serialization to match redis precision.

### PR DESCRIPTION
**Pull request by timnd**

### CODE REVIEW

File: redis.go [hunk 1]
```diff
@@ -117,29 +118,13 @@ func blocking(
...
-		// Remove trailing 0s.
-		sv = sv[:len(sv)-1]
-		// Ends with a '.'.
-		if sv[len(sv)-1] == '.' {
-			sv = sv[:len(sv)-1]
-			break
-		}
-	}
-	return sv
+	return strconv.FormatFloat(v, 'f', -1, 64)
```
Comment (alicebob) [inline redis.go]: Changing that gives me this error in miniredis_vs_redis:

```
	hash_test.go:130: value error. expected: []byte{0x2d, 0x30, 0x2e, 0x38} got: []byte{0x2d, 0x30, 0x2e, 0x37, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x39, 0x38, 0x39} case: main.command{cmd:"HINCRBYFLOAT", args:[]interface {}{"aap", "noot", -13.1}, error:false, sort:false, loosely:false}
```
That's this sequence:
```
        succ("HINCRBYFLOAT", "aap", "noot", 12.3),
        succ("HINCRBYFLOAT", "aap", "noot", -13.1),
```

Any ideas about that?

Comment (timnd) [reply redis.go]: I just took some time to look into this. It appears to me that this has to do with loss of float64 precision in:
https://github.com/alicebob/miniredis/blob/955f929b3a68c092ee308c4b340b337e759f8288/db.go#L336

My guess is that redis is doing what Go does with constants (or some approximation) which allows more precision. A solution would involve storing the string representation and doing arithmetic with added precision using `math/big`:

https://play.golang.org/p/Hg8Va7oTty0

Comment (alicebob) [reply redis.go]: That looks worth a try, since floats are kept as strings anyway. Thanks for the comment!

Comment (alicebob) [reply redis.go]: I tried that in some branch, but it had similar issues. I'll have to go read the Redis source to figure out how it is done there :(

**Pull request closed without merging**